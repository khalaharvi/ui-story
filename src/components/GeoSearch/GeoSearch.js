import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';

import connectGeoSearch from 'connectors/connectGeoSearch';
import PlaceAutocomplete from 'components/PlaceAutocomplete';

import { createClassNames } from 'utils';

const cx = createClassNames('geosearch');

export const GeoSearch = ({
  // isRefined,
  currentRefinement,
  refine,

  radiuses,
  className
}) => {
  const {
    lat,
    lng,
    name,
    radius
  } = currentRefinement;
  const disabledRadius = (!lat || !lng);

  return (
    <div className={classnames(cx(''), className)}>
      <div className={classnames(cx('select'), { [cx('select--disabled')]: disabledRadius })}>
        <select
          value={radius}
          onChange={e => refine({ lat, lng }, name, e.target.value)}
          disabled={disabledRadius}
        >
          <option value={0}>Distance</option>
          {
            radiuses.map(({ value, label }) => (
              <option key={`geosearch-radius-${value}`} value={value}>{label}</option>
            ))
          }
        </select>
      </div>
      <PlaceAutocomplete
        className={cx('autocomplete')}
        onReset={() => refine(null, null, null)}
        placeholder="Current Location"
        defaultValue={name}
        onPlaceSelected={(event, { place }) => {
          refine(place._geoloc, place.name, radius); // eslint-disable-line no-underscore-dangle
        }}
      />
    </div>
  );
};

GeoSearch.propTypes = {
  // isRefined: PropTypes.bool,
  refine: PropTypes.func.isRequired,
  currentRefinement: PropTypes.shape({
    lat: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
    lng: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
    radius: PropTypes.number
  }),
  radiuses: PropTypes.arrayOf(PropTypes.shape({
    label: PropTypes.string.isRequired,
    value: PropTypes.number.isRequired
  })),
  className: PropTypes.string
};

GeoSearch.defaultProps = {
  currentRefinement: {},
  radiuses: [{
    label: 'Within 10 miles',
    value: 10
  },
  {
    label: 'Within 25 miles',
    value: 25
  }]
};

export default connectGeoSearch(GeoSearch);
