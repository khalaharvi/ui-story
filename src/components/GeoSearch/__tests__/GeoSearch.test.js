import React from 'react';
import { shallow } from 'enzyme';

import { GeoSearch } from '../GeoSearch';

// TO DO: add more tests
describe('<GeoSearch />', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<GeoSearch refine={() => null} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('renders correctly with custom classname', () => {
    const wrapper = shallow(<GeoSearch refine={() => null} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('renders correctly with radiuses', () => {
    const radiuses = [{
      label: 'Within 1 miles',
      value: 1
    },
    {
      label: 'Within 2 miles',
      value: 2
    },
    {
      label: 'Within 3 miles',
      value: 3
    }];
    const wrapper = shallow(<GeoSearch refine={() => null} radiuses={radiuses} />);

    expect(wrapper).toMatchSnapshot();
  });

  // should call refine when onPlaceSelected
  // should call refine when onChange
  // should be disabled when not refined
});
