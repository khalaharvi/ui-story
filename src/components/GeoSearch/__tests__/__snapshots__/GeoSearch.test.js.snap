// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GeoSearch /> renders correctly 1`] = `
<div
  className="dh-geosearch"
>
  <div
    className="dh-geosearch-select dh-geosearch-select--disabled"
  >
    <select
      disabled={true}
      onChange={[Function]}
    >
      <option
        value={0}
      >
        Distance
      </option>
      <option
        key="geosearch-radius-10"
        value={10}
      >
        Within 10 miles
      </option>
      <option
        key="geosearch-radius-25"
        value={25}
      >
        Within 25 miles
      </option>
    </select>
  </div>
  <WithTheme(PlaceAutocomplete)
    className="dh-geosearch-autocomplete"
    onPlaceSelected={[Function]}
    onReset={[Function]}
    placeholder="Current Location"
  />
</div>
`;

exports[`<GeoSearch /> renders correctly with custom classname 1`] = `
<div
  className="dh-geosearch"
>
  <div
    className="dh-geosearch-select dh-geosearch-select--disabled"
  >
    <select
      disabled={true}
      onChange={[Function]}
    >
      <option
        value={0}
      >
        Distance
      </option>
      <option
        key="geosearch-radius-10"
        value={10}
      >
        Within 10 miles
      </option>
      <option
        key="geosearch-radius-25"
        value={25}
      >
        Within 25 miles
      </option>
    </select>
  </div>
  <WithTheme(PlaceAutocomplete)
    className="dh-geosearch-autocomplete"
    onPlaceSelected={[Function]}
    onReset={[Function]}
    placeholder="Current Location"
  />
</div>
`;

exports[`<GeoSearch /> renders correctly with radiuses 1`] = `
<div
  className="dh-geosearch"
>
  <div
    className="dh-geosearch-select dh-geosearch-select--disabled"
  >
    <select
      disabled={true}
      onChange={[Function]}
    >
      <option
        value={0}
      >
        Distance
      </option>
      <option
        key="geosearch-radius-1"
        value={1}
      >
        Within 1 miles
      </option>
      <option
        key="geosearch-radius-2"
        value={2}
      >
        Within 2 miles
      </option>
      <option
        key="geosearch-radius-3"
        value={3}
      >
        Within 3 miles
      </option>
    </select>
  </div>
  <WithTheme(PlaceAutocomplete)
    className="dh-geosearch-autocomplete"
    onPlaceSelected={[Function]}
    onReset={[Function]}
    placeholder="Current Location"
  />
</div>
`;
